liota/dccs/iotcc.py

```diff
@@ -109,31 +111,25 @@ def on_receive_safe(msg):
                     log.debug("Received msg: {0}".format(msg))
                     if msg != "":
                         json_msg = json.loads(msg)
-                        self.proto.on_receive(json.loads(msg))
                         log.debug("Processed msg: {0}".format(json_msg["type"]))
                         if json_msg["type"] == "create_or_find_resource_response":
                             if json_msg["body"]["uuid"] != "null":
                                 log.info("FOUND RESOURCE: {0}".format(json_msg["body"]["uuid"]))
                                 self.reg_entity_id = json_msg["body"]["uuid"]
-                                exit()
                             else:
                                 log.info("Waiting for resource creation")
-                                time.sleep(5)
-                                self.con.send(
-                                    self._registration(self.con.next_id(), entity_obj.entity_id, entity_obj.name,
-                                                       entity_obj.entity_type))
+                                self.comms.send(json.dumps(
+                                    self._registration(self.next_id(), entity_obj.entity_id, entity_obj.name,
+                                                       entity_obj.entity_type)))
+                                on_receive_safe(self.recv_msg_queue.get())
```

Comment:
I believe your assumption is that at any point in time, there is only one element in the queue, which happens to be the response for the pending request. This need not be true. For example, if one loads 10 packages each registering a different device, all requests will be from parallel threads and the queue will end up having responses for all. In such a case, this get() call will not be successful.